feat: add optional column for adapters (#419)
IKIM-Essen/uncovar: docs/user-guide/configuration.md

# Configuration

## Adapters

> There are three ways to transfer adapter sequences to UnCoVar to remove them
> from the raw data.

Reviewer: ```suggestion
There are three ways to transfer adapter sequences to UnCoVar for removing them
from the raw data.
```

### Config File

> The adapter sequences used can be entered in the config file under

Reviewer: ```suggestion
The adapter sequences used can be specified in the config file under
```

`preprocessing` -> `kit adapters`.

For **paired-end data**, the adapters can be detected by per-read overlap

> analysis, which seeks the overlap of each pair of reads. The adapter sequences

Reviewer: ```suggestion
analysis, which seeks the overlap for each pair of reads. The adapter sequences
```

> can specify the adapter sequences for read one by `—adapter_sequence` and for

Reviewer: ```suggestion
can be specified for read one by `—adapter_sequence` and for
```

> read two by`—adapter_sequence_r2`. An example is for [Illuminas TruSeq library] (<https://www.illumina.com/products/by-type/sequencing-kits/library-prep-kits/truseq-rna-v2.html>)

Reviewer: ```suggestion
read two by`—adapter_sequence_r2`. An example for [Illuminas TruSeq library] (<https://www.illumina.com/products/by-type/sequencing-kits/library-prep-kits/truseq-rna-v2.html>)
```

is shown below:

```yaml
"--adapter_sequence = AGATCGGAAGAGCACACGTCTGAACTCCAGTCA
--adapter_sequence_r2 = AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT”
```

Adapters for **single-end data** can be specified only using the
`—adapter_sequence` option.

```yaml
"--adapter_sequence = AGATCGGAAGAGCACACGTCTGAACTCCAGTCA"
```

### Sample Sheet

> The second way to remove adapter sequences is to specify the adapter sequence
> directly sample in the sample sheet. The adapters must enter it in a column

Reviewer: ```suggestion
The second way to remove adapter sequences is to specify the adapter sequence
 per sample in the sample sheet. The adapters must be entered in a column
```

called `adapters`. For paired-end and single-end format, see above. Here is
an exemplary samples sheet:

| sample_name | fq1         | fq2         | date       | is_amplicon_data | technology | adapters                                           |
|-------------|-------------|-------------|------------|------------------|------------|----------------------------------------------------|
| example-1   | PATH/TO/fq1 | PATH/TO/fq2 | 1970-01-01 | 1                | illumina   | --adapter_sequence=ACGT --adapter_sequence_r2=TGCA |
| example-2   | PATH/TO/fq  |             | 1970-01-01 | 1                | ion        | --adapter_sequence=ACGT                            |

> If an adapter sequence is entered into the sample sheet for one sample, this
> adapter sequence is obviously used to trim the sequences of this sample. For
> empty entries, UnCoVar uses the adapter sequence from the config file.

Reviewer: ```suggestion
If an adapter sequence is specified for a sample in the sample sheet, this
adapter sequence is used to trim the sequences of only this sample. For
empty entries, UnCoVar uses the adapter sequence from the config file.
```

### Pre-Defined Adapters

UnCoVar supports two different sequencing kits and their respective adapters,
namely:

1. [Revelo RNA-Seq library preparation kit](https://lifesciences.tecan.com/revelo-rna-seq-library-prep-kit?p=tab--5)
1. [EasySeq RC-PCR SARS CoV-2  Whole Genome Sequencing kit](https://www.nimagen.com/shop/products/rc-cov096/easyseq-sars-cov-2-novel-coronavirus-whole-genome-sequencing-kit)

> The `adapters` column in the sample sheet is used to trim the adapters sequences

Reviewer: ```suggestion
The `adapters` column in the sample sheet is used to trim the adapter sequences
```
